src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -2293,7 +2294,32 @@ void InterpreterSelectQuery::executeFetchColumns(QueryProcessingStage::Enum proc
         query_info.storage_limits = std::make_shared<StorageLimitsList>(storage_limits);
 
         query_info.settings_limit_offset_done = options.settings_limit_offset_done;
-        storage->read(query_plan, required_columns, storage_snapshot, query_info, context, processing_stage, max_block_size, max_streams);
+
+        /// need replay operation
+        if (settings.replay_speed != 0)
```

Comment:
`replay_speed <= 0`, means no need

Reply:
For now, we only support replay some storages:
1) StorageStream(append_only)
2) ProxyStream (proxy append_only)
Others need to throw an error `NOT_IMPLEMENTED`